test/functional/test_framework/script.py

```diff
@@ -361,8 +361,11 @@ def __init__(self, msg, data):
         self.data = data
         super(CScriptTruncatedPushDataError, self).__init__(msg)
 
+
 # This is used, eg, for blockchain heights in coinbase scripts (bip34)
-class CScriptNum():
+class CScriptNum:
+    __slots__ = ("value",)
+
     def __init__(self, d=0):
         self.value = d
 
```

Comment:
Any reason for not doing `__slots__ = ()` for `CScript` below?

Reply:
I avoided it because it used to be [documented as impossible for variable length sequences](https://docs.python.org/2.5/ref/slots.html#l2h-252). Looks like [recent revisions have been updated](https://docs.python.org/3/reference/datamodel.html#notes-on-using-slots) to clarify that it's ok as long as it's an empty `__slots__`.

I'll revise, test-to-confirm, and adjust the commit.

Reply:
Done